packages/core/test/VDocument.test.ts

```diff
@@ -1223,5 +1228,644 @@ describe('stores', () => {
+                        // Backward selection
+                        testEditor({
+                            contentBefore: '<p>]ab[cd</p>',
+                            stepFunction: enter,
+                            contentAfter: '<p><br></p><p>[]cd</p>',
+                        });
+                    });
+                    it('should delete part of a paragraph, then split it', () => {
+                        // Forward selection
+                        testEditor({
+                            contentBefore: '<p>a[bc]d</p>',
+                            stepFunction: enter,
+                            contentAfter: '<p>a</p><p>[]d</p>',
+                        });
+                        // Backward selection
+                        testEditor({
+                            contentBefore: '<p>a]bc[d</p>',
+                            stepFunction: enter,
+                            contentAfter: '<p>a</p><p>[]d</p>',
+                        });
+                    });
+                    it('should delete the last half of a paragraph, then split it', () => {
+                        // Forward selection
+                        testEditor({
+                            contentBefore: '<p>ab[cd]</p>',
+                            stepFunction: enter,
+                            contentAfter: '<p>ab</p><p>[]<br></p>',
+                        });
+                        // Backward selection
+                        testEditor({
+                            contentBefore: '<p>ab]cd[</p>',
+                            stepFunction: enter,
+                            contentAfter: '<p>ab</p><p>[]<br></p>',
+                        });
+                    });
+                    it('should delete all contents of a paragraph, then split it', () => {
+                        // Forward selection
+                        testEditor({
+                            contentBefore: '<p>[abcd]</p>',
+                            stepFunction: enter,
+                            contentAfter: '<p><br></p><p>[]<br></p>',
+                        });
+                        // Backward selection
+                        testEditor({
+                            contentBefore: '<p>]abcd[</p>',
+                            stepFunction: enter,
+                            contentAfter: '<p><br></p><p>[]<br></p>',
+                        });
+                    });
+                });
+            });
+            describe('With SHIFT key', () => {
+                describe('Range collapsed', () => {
+                    describe('Basic', () => {
+                        it('should insert a <br> into an empty paragraph', () => {
+                            testEditor({
+                                contentBefore: '<p>[]<br></p>',
+                                stepFunction: shiftEnter,
+                                // That range is equivalent to <br>[]<br>
+                                contentAfter: '<p><br><br>[]</p>',
```

Comment:
Weird, this DOM is equivalent to the "let's add an additional <br/> to make the p visible" case. But they can't be the same for the user, otherwise what happened to the br inserted by the user ?

Reply:
I don't understand your question.